ui-app/client/src/components/PageEditor/functions/EventDefGenerator.ts

```diff
@@ -0,0 +1,222 @@
+import { duplicate } from '@fincity/kirun-js';
+import { PageDefinition } from '../../../types/common';
+import { shortUUID } from '../../../util/shortUUID';
+
+interface buttonDef {
```

Comment:
There is a component definition in common.ts [ComponentDefinition]. Remove this interface.